esque/topic.py

```diff
@@ -14,6 +13,24 @@
     unpack_confluent_config,
 )
 
+class TopicConfig:
+    def __init__(self, name: str, num_partitions: int, replication_factor: int, config: Dict[str, str] = None):
+        self.name = name
+        self.num_partitions = num_partitions
+        self.replication_factor = replication_factor
+        self.config = config
+
+    def get_new_topic(self):
```

Comment:
Vendor objects should not be exposed and only used internally to ease eventual replacement of libraries. Just keep the new_topic as an instance variable and use the parent object during topic creation instead of `NewTopic`.